content/src/migrations/scripts/1664829104611_failed-deployments.ts

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { ColumnDefinitions, MigrationBuilder } from 'node-pg-migrate'
+
+export const shorthands: ColumnDefinitions | undefined = undefined
+
+export async function up(pgm: MigrationBuilder): Promise<void> {
+  pgm.createTable('failed_deployments', {
+    entity_id: { type: 'text', notNull: true, primaryKey: true },
+    entity_type: { type: 'text', notNull: true },
+    failure_time: { type: 'timestamp', notNull: true },
+    reason: { type: 'text', notNull: true },
+    auth_chain: { type: 'json', notNull: true },
+    error_description: { type: 'text', notNull: true },
+    snapshot_hash: { type: 'text', notNull: true }
```

Comment:
The failed deployments obtained while syncing from pointer changes are stored in memory, right?

Reply:
That's correct. Only failed deployments from snapshots are persisted, that's why `notNull: true`